### PR TITLE
<Leon> Finished basic challenge.

### PR DESCRIPTION
**Pull request by LeonOfTheHollow**
Changes
* * *
- [x] Refactored everything I touched to async/await instead of callbacks/thens.
* * *
To Do
* * *
1. - [ ] Build front-end
2. - [ ] Extinguish callbacks from src
3. - [ ] Style front-end
4. - [ ] Bonus features

### CODE REVIEW

**Comment by mixelpixel** [comment]
I ❤️ your first PR comment organization. I edited it with the GH Task List syntax for checking items off and displaying a status bar in the main PR list for this repo:

screen shot 2018-03-25 at 2 00 06 pm

...also ❤️❤️❤️ that you are using the Git Commit formatting of title and body 👍

**Comment by mixelpixel** [comment]
NOTE: I have you with two GH handles: IchabodPain and LeonOfTheHollow - fine either way, but are these both active, or did you change from Ichabod to Hollow?

**Comment by mixelpixel** [comment]
Great answers to the study questions.

I "get" the dad jokes from your server...

File: api/controllers/login.js [hunk 1]
```diff
@@ -1,33 +1,56 @@
...
+
+const login = async function(req, res) {
   const { username, password } = req.body;
-  User.findOne({ username }, (err, user) => {
-    if (err) {
-      res.status(403).json({ error: 'Invalid Username/Password' });
-      return;
-    }
+  const formattedUsername = username.toLowerCase();
+  console.log(formattedUsername);
```
Comment (mixelpixel) [inline api/controllers/login.js]: useful in development, but these kinds of console.log()s can be removed.

**Comment by mixelpixel** [comment]
does the "killer-jokes" front end app need to be in a sub directory?

File: front-end/killer-jokes/src/index.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  BrowserRouter as Router,
+  Route,
+} from 'react-router-dom';
+import reducers from './reducers';
+import './index.css';
+
+const createStoreWithMiddleware = applyMiddleware(ReduxThunk)(createStore);
+
+ReactDOM.render(
+  <Provider store={createStoreWithMiddleware(reducers)}>
```
Comment (mixelpixel) [inline front-end/killer-jokes/src/index.js]: If you change line 22 to:
```js
<Provider store={createStoreWithMiddleware(reducers, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__())}>
```

...then you could make use of the Redux DevTools plugin:

screen shot 2018-03-25 at 2 39 54 pm

**Comment by LeonOfTheHollow** [comment]
I just changed it to the second one recently. I felt it was a bit more
professional.

**Pull request closed without merging**